keyhive_core/src/keyhive.rs

```diff
@@ -1428,7 +1428,7 @@ impl<
             );
         }
 
-        for (digest, static_op) in archive.topsorted_ops.iter() {
+        for (digest, static_op) in archive.topsorted_ops.iter().rev() {
```

Comment:
Oh yeah, I think I know how this happened; we used to produce things reverse order in the topsort (which was confusing) and I think this `pop_all` _when we create_ the archive is the actual culprit:

```rust
        while !adjacencies.is_empty() {
            let mut latest = adjacencies.pop_all(); // ⬅️😈

            // NOTE sort all concurrent heads by hash for more determinism
            latest.sort_by_key(|(digest, _)| *digest);

            for (digest, op) in latest.into_iter() {
                dependencies.push((digest, op.clone()));
            }
        }

```